### PR TITLE
add(app): mempool

### PR DESCRIPTION
**Pull request by MontejoJorge**
## Summary by CodeRabbit

- **New Features**
  - Introduced configuration options for the Mempool application, including settings for Bitcoin node and Electrs host addresses, ports, data folder location, and Mempool DB password.
  - Added a `docker-compose` setup for Mempool, including services for `mempool`, `mempool-api`, and `mempool-db` to streamline container management.
  - Included detailed project description highlighting Mempool's role in visualizing and exploring the Bitcoin transaction fee market.

### CODE REVIEW

File: apps/mempool/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    container_name: mempool-db
+    image: mariadb:10.5.21
+    restart: unless-stopped
+    stop_grace_period: 1m
+    user: "1000:1000"
+    environment:
+      MYSQL_DATABASE: "mempool"
+      MYSQL_USER: "mempool"
+      MYSQL_PASSWORD: "mempool"
+      MYSQL_ROOT_PASSWORD: "admin"
```
[the remark this answers is not in the document]
Comment (MontejoJorge) [reply apps/mempool/docker-compose.yml:85]: The container is not exposed, no need to do it

[the remark this answers is not in the document]
Comment (MontejoJorge) [reply apps/mempool/docker-compose.yml]: done

File: apps/mempool/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    volumes:
+      - ${BITCOIND_DIR:-${APP_DATA_DIR}/../bitcoind/data}:/data/.bitcoin:ro
+      - ${APP_DATA_DIR}/data:/backend/cache
+    command: "./start.sh"
+    depends_on:
+      - mempool-db
+    environment:
+      MEMPOOL_BACKEND: "electrum"
+      ELECTRUM_HOST: "electrs"
+      ELECTRUM_PORT: "50001"
```
Comment (DireMunchkin) [inline apps/mempool/docker-compose.yml:60]: You could make the Electrs host / port configurable like with the Bitcoin host, and default to the current values.

Comment (MontejoJorge) [reply apps/mempool/docker-compose.yml]: Sure!

**Review by DireMunchkin — APPROVED** [review]

File: apps/mempool/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
+services:
+  mempool:
+    container_name: mempool
+    image: mempool/frontend:v3.0.0-dev2
+    restart: unless-stopped
+    stop_grace_period: 1m
+    user: 0:0
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:7]: should be avoided by default

File: apps/mempool/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    command: "nginx -g 'daemon off;'"
+    ports:
+      - 2200:8080
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:10]: shoud use var port instead

File: apps/mempool/config.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+   "$schema": "../schema.json",
+   "name": "Mempool",
+   "available": true,
+   "exposable": false,
+   "no_gui": false,
+   "port": 2200,
+   "id": "mempool",
+   "description": "Bitcoin mempool visualizer and blockchain explorer",
+   "tipi_version": 1,
+   "version": "v3.0.0-dev2",
```
Comment (JigSawFr) [inline apps/mempool/config.json:11]: all versions should be stable (but can be overrided by user-config)

File: apps/mempool/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+  mempool-api:
+    container_name: mempool-api
+    image: mempool/backend:v3.0.0-dev2
+    restart: unless-stopped
+    stop_grace_period: 1m
+    user: 0:0
+    volumes:
+      - ${BITCOIND_DIR:-${APP_DATA_DIR}/../bitcoind/data}:/data/.bitcoin:ro
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:52]: is it working as expected ?

File: apps/mempool/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      traefik.http.routers.mempool-local.service: mempool
+      traefik.http.routers.mempool-local.tls: true
+
+
+  mempool-api:
+    container_name: mempool-api
+    image: mempool/backend:v3.0.0-dev2
+    restart: unless-stopped
+    stop_grace_period: 1m
+    user: 0:0
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:50]: should be avoided

File: apps/mempool/docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      # Local domain secure
+      traefik.http.routers.mempool-local.rule: Host(`mempool.${LOCAL_DOMAIN}`)
+      traefik.http.routers.mempool-local.entrypoints: websecure
+      traefik.http.routers.mempool-local.service: mempool
+      traefik.http.routers.mempool-local.tls: true
+
+
+  mempool-api:
+    container_name: mempool-api
+    image: mempool/backend:v3.0.0-dev2
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:47]: need stable version

File: apps/mempool/docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+      STATISTICS_ENABLED: "true"
+    networks:
+      - tipi_main_network
+
+  mempool-db:
+    container_name: mempool-db
+    image: mariadb:10.5.21
+    restart: unless-stopped
+    stop_grace_period: 1m
+    user: 0:0
```
Comment (JigSawFr) [inline apps/mempool/docker-compose.yml:81]: should be avoided

**Review by JigSawFr — CHANGES_REQUESTED** [review]

File: apps/mempool/docker-compose.yml (hunk 6 above)
Comment (DireMunchkin) [reply apps/mempool/docker-compose.yml:52]: I've tested this out, it works as expected. Either the value of `BITCOIND_DIR` is used if set, or by default your `app-data/bitcoind/data` folder is used.

File: apps/mempool/config.json (hunk 5 above)
Comment (DireMunchkin) [reply apps/mempool/config.json:11]: The latest stable version is v2.5.0, can use that image version instead (same for the other one)

Comment (JigSawFr) [reply apps/mempool/config.json:11]: Yes please

Comment (MontejoJorge) [reply apps/mempool/config.json:11]: Im using this version because the 2.5.0 does not have suport for the .cookie auth method

Comment (DireMunchkin) [reply apps/mempool/config.json:11]: That's fair enough for me, @JigSawFr would you be OK with keeping the existing image?

Comment (DireMunchkin) [reply apps/mempool/config.json:11]: @JigSawFr  @MontejoJorge Do you think we could get this PR unstuck? Last I checked there was some question about if it was OK to use the non stable image. Can we come to a conclusion on that?

Comment (MontejoJorge) [reply apps/mempool/config.json:11]: I cant decide about that, @JigSawFr @meienberger @steveiliop56 Any thoughts about this?

Comment (DireMunchkin) [reply apps/mempool/config.json:11]: If you wait long enough the Mempool guys will release 3.0.0 , they just tagged [`beta1`](https://github.com/mempool/mempool/releases/tag/v3.0.0-beta1) 😁

Comment (MontejoJorge) [reply apps/mempool/config.json]: @DireMunchkin [https://github.com/mempool/mempool/releases/tag/v3.0.0-rc1](rc1) is out, so comming soon i guess

Comment (DireMunchkin) [reply apps/mempool/config.json]: It's out now: https://github.com/mempool/mempool/releases/tag/v3.0.0

File: apps/mempool/config.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
+{
+   "$schema": "../schema.json",
+   "name": "Mempool",
+   "available": true,
+   "exposable": false,
```
Comment (meienberger) [inline apps/mempool/config.json:5]: Any reason you marked it as non-exposable? I think it should be

**Review by meienberger — CHANGES_REQUESTED** [review]

File: apps/mempool/config.json (hunk 10 above)
Comment (MontejoJorge) [reply apps/mempool/config.json:5]: You are right

**Review by meienberger — APPROVED** [review]

**Pull request merged**